generator/src/test/java/com/scottlogic/deg/generator/restrictions/NumericRestrictionsMergerTests.java

```diff
@@ -102,4 +102,62 @@ public void merge_withContradictoryNumericRestrictions_shouldReturnUnsuccessful(
         Assert.assertThat(result, not(nullValue()));
         Assert.assertThat(result.successful, is(false));
     }
+
+    @Test
+    public void merge_withScaleEqualToRange_shouldReturnSuccessful(){
+        NumericRestrictionsMerger merger = new NumericRestrictionsMerger();
+        NumericRestrictions left = new NumericRestrictions();
+        left.min = new NumericLimit<>(BigDecimal.ZERO, true);
+        left.max = new NumericLimit<>(BigDecimal.ONE, true);
+        NumericRestrictions right = new NumericRestrictions(0);
+
+        MergeResult<NumericRestrictions> result = merger.merge(left, right);
+
+        Assert.assertThat(result, not(nullValue()));
+        Assert.assertThat(result.successful, is(true));
+    }
+
+    @Test
+    public void merge_withScaleEqualToRangeExclusiveMax_shouldReturnSuccessful(){
+        NumericRestrictionsMerger merger = new NumericRestrictionsMerger();
+        NumericRestrictions left = new NumericRestrictions();
+        left.min = new NumericLimit<>(BigDecimal.ZERO, true);
+        left.max = new NumericLimit<>(BigDecimal.ONE, false);
+        NumericRestrictions right = new NumericRestrictions(0);
+
+        MergeResult<NumericRestrictions> result = merger.merge(left, right);
+
+        Assert.assertThat(result, not(nullValue()));
+        Assert.assertThat(result.successful, is(true));
+    }
+
+    @Test
+    public void merge_withScaleEqualToRangeeExclusiveMin_shouldReturnSuccessful(){
+        NumericRestrictionsMerger merger = new NumericRestrictionsMerger();
+        NumericRestrictions left = new NumericRestrictions();
+        left.min = new NumericLimit<>(BigDecimal.ZERO, false);
+        left.max = new NumericLimit<>(BigDecimal.ONE, true);
+        NumericRestrictions right = new NumericRestrictions(0);
+
+        MergeResult<NumericRestrictions> result = merger.merge(left, right);
+
+        Assert.assertThat(result, not(nullValue()));
+        Assert.assertThat(result.successful, is(true));
+    }
+
+    @Test
+    public void merge_withScaleLargerThan_shouldReturnUnsuccessful(){
+        NumericRestrictionsMerger merger = new NumericRestrictionsMerger();
+        NumericRestrictions left = new NumericRestrictions();
+        left.min = new NumericLimit<>(BigDecimal.ZERO, false);
+        left.max = new NumericLimit<>(BigDecimal.ONE, false);
+        NumericRestrictions right = new NumericRestrictions(0);
+
+        MergeResult<NumericRestrictions> result = merger.merge(left, right);
+
+        Assert.assertThat(result, not(nullValue()));
+        Assert.assertThat(result.successful, is(false));
+    }
+
+
```

Comment:
There should be a couple of tests, at least, for when the granularity is different, i.e. i presume all of the above cover integer granularity (however its difficult to tell as the granularity is 0?). There should be some to prove that if the granularity is - say 0.1 or 0.01 - then the state is computed correctly

Reply:
0 represents the scale not the granularity. Nevertheless there should be some tests where this is positive.

- that it merges correctly, uses the lowest scale (still)
- that it correctly identifies if there are values permitted between min and max given the granularity